ppl/zqd/recruiter.go

```diff
@@ -32,6 +23,8 @@ func handleRecruit(c *Core, w http.ResponseWriter, r *http.Request) {
 	}
 	workers := make([]api.Worker, len(ws))
 	for i, e := range ws {
+		// Note the receiver for this channel is the select in handleRegister below.
```

Comment:
There's an issue here: the handleRegister timeout could fire just before a worker is selected for recruitment, which would leave this request stuck trying to write to a channel that has no listeners.

Reply:
This is a really good point. I cannot do a blocking write here because it could cause the call to /recruiter/recruit to hang. In my latest commit, I converted this to a non-blocking write to the "recruited" receiver. My intent here is in the event that the /recruit and the registration timeout are nearly simultaneous, I want the registration timeout to win, and the recruit to return fewer workers. I added logic to do that by having a callback in the recruiter.WorkerDetail struct, and using that callback to do a non-blocking write to the "recruited" receiver. In addition, I "Stop()" the timer in the callback to narrow the window of this race condition. (I read the code for timer.Stop(), and it looks like it is safe to call from another thread.)